vault/core_test.go

```diff
@@ -807,6 +807,106 @@ func TestCore_HandleRequest_AuditTrail(t *testing.T) {
 	}
 }
 
+func TestCore_HandleRequest_AuditTrail_noHMACKeys(t *testing.T) {
```

Comment:
I am not sure if this test case does what you intend it to do. I mean NoopAudit just copies the input to its output. It doesn't hash anything. You could provide an input that ideally should be obfuscated and see that it doesn't. Was there a different intention to this test? If yes, can we please document that. If not, probably we need to think about a different way to test this out.

Reply:
`NoopAudit` got extended to include `ReqNonHMACKeys` `RespNonHMACKeys` and `RespReqNonHMACKeys` as part of things that it tracks, so this test makes sure that non-HMAC values are passed into the backends. This test does not actually test for the correctness of the hashing, but rather the flow of the values, from the point where they are specified via mount tune, all the way into the audit's backend.

The testing of the hashing function is done in audit/hashstructure_test.go